### PR TITLE
First round of cleanup and having strong analyzer for project

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
Description

- Improves couple of test cases
- Uses preferred final variables
- Project had lots of unused imports in all files due to analyzer disabled for it, fixes that .
- Removes unused local variables (can remove from analyzer, but there's a unused local variable for did in project which will make analyzer test unhappy, could remove in second round)
- Updates readme

Update:
- Have completed a todo and have enabled the analyzer to check for unused local variables

### CODE REVIEW

**Comment by AyushBherwani1998** [comment]
@Plumillon is this package inspired from JS SDK of PolygonId? There are quite a few Todos and could contribute to fix it. The issue is the code is not documented that much and it gets little difficult to understand. The JS SDK is well documented, if it's inspired from the JS SDK, I can go through the documentation and flow and try to fix Todos.

**Comment by mmilenkovic** [comment]
> @Plumillon is this package inspired from JS SDK of PolygonId? There are quite a few Todos and could contribute to fix it. The issue is the code is not documented that much and it gets little difficult to understand. The JS SDK is well documented, if it's inspired from the JS SDK, I can go through the documentation and flow and try to fix Todos.

First thanks @AyushBherwani1998 for the interest in improving our repo. To answer your question this package is not inspired by JS SDK of PolygonId. We have different solutions for web and native wallets. We are indeed trying to improve our docs and will check your PR thoroughly. Having that said our main documentation sits at https://github.com/0xPolygonID/tutorials (https://0xpolygonid.github.io/tutorials/) where you have Flutter SDK section.

Let's continue the conversation on how we can improve and thanks again for contributing!

File: lib/credential/domain/use_cases/get_claims_use_case.dart [hunk 1]
```diff
@@ -36,22 +36,25 @@ class GetClaimsUseCase
...
-              genesisDid: param.genesisDid,
-              privateKey: param.privateKey)
-          .then((claims) {
-        logger().i("[GetClaimsUseCase] Claims: $claims");
-        return claims;
-      }).catchError((error) {
-        logger().e("[GetClaimsUseCase] Error: $error");
-        throw error;
-      });
+      if (param.genesisDid == did) {
```
Comment (Plumillon) [inline lib/credential/domain/use_cases/get_claims_use_case.dart:42]: This check is not valid in this case (the TODO was deprecated and misleading), but it raised a concern about the check and I fixed it here: https://github.com/0xPolygonID/polygonid-flutter-sdk/pull/315
Can you revert this file in this PR?

**Review by Plumillon — CHANGES_REQUESTED** [review]

File: lib/credential/domain/use_cases/get_claims_use_case.dart (hunk 1 above)
Comment (AyushBherwani1998) [reply lib/credential/domain/use_cases/get_claims_use_case.dart]: @Plumillon got it, have reverted the change and local unused variable did to make analyzer happy.

**Comment by AyushBherwani1998** [comment]
@mmilenkovic @Plumillon do you folks have any discord group where I can ask questions regarding the SDK? I'll be diving deep and help contribute in any way I can. It would be helpful to get guidance from you folks if stuck somewhere.

**Comment by AyushBherwani1998** [comment]
@Plumillon okay, I'm already a member of that group. I'll ask there or create a discussion. Thanks.

File: lib/credential/domain/use_cases/get_claims_use_case.dart (hunk 1 above)
Comment (AyushBherwani1998) [reply lib/credential/domain/use_cases/get_claims_use_case.dart]: @Plumillon hope you are doing fine, can you PTAL?

Comment (Plumillon) [reply lib/credential/domain/use_cases/get_claims_use_case.dart]: Hey @AyushBherwani1998!
I'm not a maintainer anymore, please refer to @rauljareno and @plusema86 :)

**Pull request closed without merging**